File: saferide-backend-python/models/responses/ride_response.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime

class RideResponse(BaseModel):
    ride_id: str = Field(..., description="Unique ride identifier")
    status: str = Field(..., description="Current ride status")
    driver_info: Optional[Dict[str, Any]] = Field(None, description="Driver information if assigned")
    estimated_pickup: datetime = Field(..., description="Estimated pickup time")
    estimated_arrival: datetime = Field(..., description="Estimated arrival time")
    fare_estimate: float = Field(..., description="Estimated fare amount") 

File: saferide-frontend/src/models/responses/ride_response.ts
/**
 * Driver information interface
 */
export interface DriverInfo {
  id: string
  name: string
  rating: number
  car: string
}

/**
 * Ride response interface for ride booking results
 */
export interface RideResponse {
  ride_id: string
  status: string
  driver_info?: DriverInfo
  estimated_pickup: string
  estimated_arrival: string
  fare_estimate: number
} 